File: spec/factories/invoice.rb
FactoryBot.define do
  factory :invoice do
    association :user
    association :client
    association :currency

    name { Faker::Name.name }
    issue_date { Time.current - 2.days }
    due_date { Time.current - 10.days }
    status { Invoice.statuses.keys.sample }
    total_price { 100.59 }
  end

  factory :draft_invoice, class: 'Invoice' do
    association :user
    status { 'draft' }
  end
end

File: app/packs/javascript/models/Invoice.ts
import { Client } from './Client';
import { Currency } from './Currency';
import { User } from './User';
import { InvoiceItem } from './InvoiceItem';

export type Status = 'draft' | 'pending' | 'paid'
export interface Invoice {
  id?: string;
  uuid?: string;
  name?: string;
  issueDate?: Date;
  dueDate?: Date;
  totalPrice?: number;
  status?: Status;
  client?: Client;
  user?: User;
  currency?: Currency;
  createdAt?: Date;
  updatedAt?: Date;
  items?: InvoiceItem[]
}
